recOrder/cli/compute_transfer_function.py

```diff
@@ -34,76 +45,10 @@ def compute_transfer_function_cli(config_path, output_path):
-
-        # Calculate transfer functions
-        intensity_to_stokes_matrix = (
-            inplane_anisotropic_thin_pol3d.calculate_transfer_function(
-                **settings.birefringence_transfer_function_settings.dict()
-            )
-        )
-        # Save
-        dataset[
-            "intensity_to_stokes_matrix"
-        ] = intensity_to_stokes_matrix.cpu().numpy()[None, None, None, ...]
+        dataset = generate_birefringence(settings, dataset)
 
     if settings.universal_settings.reconstruct_phase:
-        echo_headline("Generating phase transfer function with settings:")
-        echo_settings(settings.phase_transfer_function_settings)
-
-        if settings.universal_settings.reconstruction_dimension == 2:
-            # Convert zyx_shape and z_pixel_size into yx_shape and z_position_list
-            settings_dict = settings.phase_transfer_function_settings.dict()
-            z_shape, y_shape, x_shape = settings_dict["zyx_shape"]
-            settings_dict["yx_shape"] = [y_shape, x_shape]
-            settings_dict["z_position_list"] = list(
-                -(np.arange(z_shape) - z_shape // 2)
-                * settings_dict["z_pixel_size"]
-            )
-
-            # Remove unused parameters
-            settings_dict.pop("zyx_shape")
-            settings_dict.pop("z_pixel_size")
-            settings_dict.pop("z_padding")
-
-            # Calculate transfer functions
-            (
-                absorption_transfer_function,
-                phase_transfer_function,
-            ) = isotropic_thin_3d.calculate_transfer_function(
-                **settings_dict,
-                wavelength_illumination=settings.universal_settings.wavelength_illumination,
-            )
-
-            # Save
-            dataset[
-                "absorption_transfer_function"
-            ] = absorption_transfer_function.cpu().numpy()[None, None, ...]
-            dataset[
-                "phase_transfer_function"
-            ] = phase_transfer_function.cpu().numpy()[None, None, ...]
-
-        elif settings.universal_settings.reconstruction_dimension == 3:
-            # Calculate transfer functions
-            (
-                real_potential_transfer_function,
-                imaginary_potential_transfer_function,
-            ) = phase_thick_3d.calculate_transfer_function(
-                **settings.phase_transfer_function_settings.dict(),
-                wavelength_illumination=settings.universal_settings.wavelength_illumination,
-            )
-            # Save
-            dataset[
-                "real_potential_transfer_function"
-            ] = real_potential_transfer_function.cpu().numpy()[None, None, ...]
-            dataset[
-                "imaginary_potential_transfer_function"
-            ] = imaginary_potential_transfer_function.cpu().numpy()[
-                None, None, ...
-            ]
+        datatset = generate_phase(settings, dataset)
```

Comment:
See me comment above...the typo isn't actually a bug because you don't need to return dataset. 

I'll also suggest `generate_phase_transfer_function`.

Reply:
I removed the return, and renamed the functions to generate_and_save_phase_transfer_function and generate_and_save_birefringence_transfer_function.